src/classes/RD_RecurringDonations_BATCH.cls

```diff
@@ -93,16 +93,21 @@ Schedulable {
         // this must return a QueryLocator, or else uncaught exceptions in the
         // execute() method will not properly update the processed/failure
         // counts (see W-2443023).
-        return Database.getQueryLocator(query);      
+
+        if (RD2_EnablementService.isRecurringDonations2Enabled) {
```

Comment:
If we are stopping legacy RD batch to execute when enhanced RDs are enabled, then code above
```
if (RD2_EnablementService.isRecurringDonations2Enabled) {
    query += ' LIMIT 0';
}
```
is not valid any longer.

Reply:
Good point